pages/airdrops/create.tsx

```diff
@@ -289,209 +321,235 @@ const CreateAirdrop: NextPage = () => {
+                  'before:absolute before:inset-0 before:hover:bg-white/5 before:transition'
+                )}
+                onChange={onFileChange}
+                ref={inputFile}
+                type="file"
+              />
+            </div>
           )}
-          {expirationType == 'timestamp' && (
-            <DateTimePicker
-              className={clsx(
-                'bg-white/10 rounded border-2 border-white/20 form-input',
-                'placeholder:text-white/50',
-                'focus:ring focus:ring-plumbus-20'
+          {accountsFile && (
+            <div className="flex flex-col bg-stone-800/80 rounded border-2 border-white/20">
+              <div className="flex justify-center py-2 px-4 space-x-2 border-b-2 border-white/20">
+                <span className="font-mono">{accountsFile.name}</span>
+                <button
+                  className="flex items-center text-plumbus hover:text-plumbus-light rounded-full"
+                  onClick={removeFileOnClick}
+                >
+                  <IoCloseSharp size={22} />
+                </button>
+              </div>
+              {fileContents && (
+                <div className="overflow-auto p-2 h-[400px] font-mono text-sm hover:resize-y">
+                  <pre>{JSON.stringify(fileContents, null, 2).trim()}</pre>
+                </div>
               )}
-              onChange={(date) => setExpirationDate(date)}
-              minDate={new Date()}
-              value={expirationDate ?? undefined}
-            />
+            </div>
           )}
-        </div>
+        </FormControl>
       </div>
 
-      {accountsFile && (
-        <div className="flex justify-center items-center text-lg font-bold">
-          Selected file name: {accountsFile.name}{' '}
-          <IoCloseSharp
-            onClick={removeFileOnClick}
-            className="ml-1 w-5 h-5 cursor-pointer"
-          />
-        </div>
-      )}
-
-      {fileContents && (
-        <SyntaxHighlighter
-          language="javascript"
-          style={prism}
-          customStyle={{ maxHeight: 470, height: 470 }}
-        >
-          {`${JSON.stringify(fileContents.slice(0, 4), null, 2)}${
-            fileContents.length > 4 ? '...and more' : ''
-          }`}
-        </SyntaxHighlighter>
-      )}
-
-      <input
-        accept=".csv"
-        id="file"
-        onChange={onFileChange}
-        ref={inputFile}
-        style={{ display: 'none' }}
-        type="file"
-      />
-
-      <br />
-
-      <button
-        className={clsx(
-          'py-2 px-4 w-full font-bold',
-          'bg-plumbus-60 hover:bg-plumbus-50 rounded focus:ring',
-          { 'opacity-50 cursor-wait': loading }
-        )}
-        disabled={loading}
-        onClick={uploadJSONOnClick}
+      <div
+        className={clsx('flex justify-end pb-6', {
+          'sticky right-0 bottom-0': isValidToCreate,
+        })}
```

Comment:
I don't see anything wrong with the logic, but breaking this form up into section components would be very helpful for readability. As it stands, the file is >500 lines long

Reply:
If we split these, prop drilling will be required to pass all form states and handlers. Again, we can iterate further more until we split all business logics.